Integrations/Kafka_V2/Kafka_V2.py

```diff
@@ -316,75 +332,121 @@ def fetch_partitions():
         return_error('Topic {} was not found in Kafka'.format(topic))
 
 
-def fetch_incidents():
+def fetch_incidents(client):
     """
     Fetches incidents
     """
-    incidents = list()
-
-    # Check for topic in Kafka
-    if TOPIC in KAFKA_CLIENT.topics:
-        kafka_topic = KAFKA_CLIENT.topics[TOPIC]
-        offset, partition = check_params(kafka_topic, old_offset=OFFSET, old_partition=PARTITION)
-        last_offset = demisto.getLastRun().get('last_offset')
-        last_fetch = last_offset if last_offset > offset else offset
-
-        # If need to fetch
-        latest_offset = check_latest_offset(kafka_topic, partition)
-        if latest_offset > last_fetch:
-            consumer = kafka_topic.get_simple_consumer(
-                auto_offset_reset=last_fetch,
-                reset_offset_on_start=True
-            )
-            for i in range(last_fetch, latest_offset):
-                """
-                consumer.consume() will consume only one message from the given offset.
-                """
-                message = consumer.consume()
-                incidents.append(create_incident(message=message, topic=TOPIC))
-                if message.offset == latest_offset:
+    topics_to_fetch_from = argToList(demisto.params().get('topic', ''))
+    partition_to_fetch_from = argToList(demisto.params().get('partition', ''))
+    offset_to_fetch_from = demisto.params().get('offset', -2)
+    try:
+        offset_to_fetch_from = int(offset_to_fetch_from)
+    except ValueError:
+        offset_to_fetch_from = -2
+    max_messages = demisto.params().get('max_messages', 50)
+    try:
+        max_messages = int(max_messages)
+    except ValueError:
+        max_messages = 50
+
+    last_fetched_partitions_offset = json.loads(demisto.getLastRun().get('last_fetched_partitions_offset', '{}'))
+
+    incidents = []
+
+    message_counter = 0
+
+    for topic in client.topics.values():
+        if (topics_to_fetch_from and topic.name in topics_to_fetch_from) or not topics_to_fetch_from:
```

Comment:
If I understand correctly if topic_to_fetch_from is empty we fetch from all topics. Sounds to me extreme. Let's always require at least one topic defined.